apis/database/v1alpha1/redis/redis_types.go

```diff
@@ -0,0 +1,126 @@
+/*
```

Comment:
Whether should `redis` be located in `apis/database`? I think it's better to be in a separate one `apis/redis`.

I look through apis of provider aws/azure/gcp, it seems all managed resource apis are located in isolated directory. Besides, isolating api will help this managed to be integrated in a scenario that only redis is to be integrated without rds.